src/Error/TypeError.hs

```diff
@@ -34,4 +35,5 @@ instance Show TypeError where
   show (BadApp n e)              = "Could not apply " ++ n ++ " to " ++ show e ++ "; it is not a function."
   show (Dereff n _t)             = "Could not dereference the function " ++ n ++ " with type " ++ show _t ++ ". Maybe you forgot to give it arguments."
   show (Uninitialized n)         = "Incomplete initialization of Board " ++ quote n
-  show (SigBadFeq n sig f)    = quote (n ++ " : " ++ show sig) ++ " cannot be defined with the function equation\n\t" ++ show f
+  show (SigBadFeq n sig f)       = quote (n ++ " : " ++ show sig) ++ " cannot be defined with the function equation\n\t" ++ show f
+  show (InputMismatch act xp e)  = "Got type " ++ show act ++ " but expected type " ++ show xp ++ " from input:\n\t" ++ show e
```

Comment:
A small punctuation nit, can add a comma between got & expected. `Got type Int, but expected...`

Reply:
Good catch; I will fix it.